src/test/tv/codely/scala_intro_examples/lesson_09_oop/exercise_solutions/joan_miralles_solution/EmailTest.scala

```diff
@@ -0,0 +1,46 @@
+package tv.codely.scala_intro_examples.lesson_09_oop.exercise_solutions.joan_miralles_solution
+
+import java.security.InvalidParameterException
+
+import org.scalatest.{Matchers, WordSpec}
+
+class EmailTest extends WordSpec with Matchers {
+
+  private val validEmail = "some@email.com"
+  private val anotherValidEmail = "a@e.t"
+  private val invalidEmail = "some.email.com"
+  private val anotherInvalidEmail = "some@email"
```

Comment:
Since these values are only used in a single test case, maybe we could define them inside the test case body:

```scala
"return a new Email instance given a complete email (2)" in {
  val anotherInvalidEmail = "some@email"
  val email = Email(anotherValidEmail)
  assert(email.isInstanceOf[Email])
  email.local shouldBe "a"
  email.domain shouldBe "e.t"
}
```